beetmoverscript/src/beetmoverscript/gcloud.py

```diff
@@ -101,12 +119,74 @@ async def upload_to_gcs(context, target_path, path):
     return blob.upload_from_filename(path, content_type=mime_type)
 
 
+def build_artifact_registry_gcs_source(context, product):
+    bucket_name = get_bucket_name(context, product, "gcloud")
+    gcs_source_kwargs = {
+        "uris": [f"gs://{bucket_name}/{gcs_source}" for gcs_source in context.task["payload"]["gcs_sources"]],
+        "use_wildcards": False,
+    }
+    if context.resource_type == "apt-repo":
+        return artifactregistry_v1.ImportAptArtifactsGcsSource(**gcs_source_kwargs)
+    if context.resource_type == "yum-repo":
+        return artifactregistry_v1.ImportYumArtifactsGcsSource(**gcs_source_kwargs)
+    raise Exception("Artifact Registry resource must be one of [apt-repo, yum-repo]")
+
+
+def build_artifact_registry_import_artifacts_request(context, repository, gcs_source):
+    import_request_kwargs = {
+        "gcs_source": gcs_source,
+        "parent": repository.name,
+    }
+    if context.resource_type == "apt-repo":
+        return artifactregistry_v1.ImportAptArtifactsRequest(**import_request_kwargs)
+    if context.resource_type == "yum-repo":
+        return artifactregistry_v1.ImportYumArtifactsRequest(**import_request_kwargs)
+    # artifact registry supports docker... interesting...
+    raise Exception("Artifact Registry resource must be one of [apt-repo, yum-repo]")
+
+
+def do_artifact_registry_import_artifacts_request(context, import_artifacts_request):
+    if context.resource_type == "apt-repo":
+        return context.gcp_client.import_apt_artifacts(request=import_artifacts_request)
+    if context.resource_type == "yum-repo":
+        return context.gcp_client.import_yum_artifacts(request=import_artifacts_request)
+    raise Exception("Artifact Registry resource must be one of [apt-repo, yum-repo]")
+
+
+async def import_from_gcs_to_artifact_registry(context):
+    """Imports release artifacts from gcp cloud storage to gcp artifact registry"""
+    product = get_product_name(context.task, context.config)
+    project = get_resource_project(context, product, "gcloud")
+    location = get_resource_location(context, product, "gcloud")
+    repository_name = get_resource_name(context, product, "gcloud")
+    parent = f"projects/{project}/locations/{location}/repositories/{repository_name}"
+    get_repo_request = artifactregistry_v1.GetRepositoryRequest(
+        name=parent,
+    )
+
+    repository = await context.gcp_client.get_repository(request=get_repo_request)
+    log.info(repository)
+
+    gcs_source = build_artifact_registry_gcs_source(context, product)
+    log.info(gcs_source)
+
+    import_artifacts_request = build_artifact_registry_import_artifacts_request(context, repository, gcs_source)
+    log.info(import_artifacts_request)
+
+    async_operation = await do_artifact_registry_import_artifacts_request(context, import_artifacts_request)
```

Comment:
Looking at `build_artifact_registry_import_artifacts_request`, `build_artifact_registry_gcs_source` and `do_artifact_registry_import_artifacts_request`, makes me wish we were more object-oriented (with classes).
All 3 functions are switching on the same flag, and "doing more than one thing".

One way of simplifying  the logic would be to have the switching in `import_from_gcs_to_artifact_registry`
```
if context.resource_type == "apt-repo":
    result = import_apt(....)
elif context.resource_type == "yum-repo":
    result = import_yum(...)
else:
    raise Exception("Artifact Registry resource must be one of [apt-repo, yum-repo]")
```